mpas_analysis/analysis_task_template.py

```diff
@@ -0,0 +1,227 @@
+        #     the folder where the task resides)
+        # tags are some useful names describing the task ('timeSeries',
+        #     'climatology', 'horizontalMap', 'index', 'transect', etc.) that
+        #     can be used in the 'generate' config option, e.g.
+        #     'all_climatology' of 'no_transect'.  Tasks that include computing
+        # time series, indices and/or climatologies should include
+        # 'timeSeries', 'index' and/or 'climatology' tags, as these are helpful
+        # used to compute and update the start and end dates of each of these
+        # analysis tasks as part of setting up the task.
+        super(MyTask, self).__init__(config=config,
+                                     taskName='myTask',
+                                     componentName='component',
+                                     tags=['tag1', 'tag2'])
+
+        # then, store any additional arguments for use later on.  These would
+        # likely include things like the name of a field, region, month,
+        # season, etc. to be analyzed so that the same subclass of AnalysisTask
+        # can perform several different tasks (potentially in parallel)
+        # Modify or remove the folowing:
+        self.myArg = myArg
+        # }}}
+
+    # this function will be called to figure out if the analysis task should
+    # run.  It shoul check if the input arguments to the task are supported,
+    # if appropriate analysis member(s) (AMs) were turned on, if the necessary
+    # output files exist, etc.  If there is a problem, an exception should be
+    # raised (see the example below) so the task will not be run.
+    def setup_and_check(self):  # {{{
+        '''
+        Perform steps to set up the analysis and check for errors in the setup.
+
+        Raises
+        ------
+        ValueError: <if myArg has an invalid value; modify as needed>
+
+        Authors
+        -------
+        <List of authors>
+        '''
+
+        # first, call setup_and_check from the base class (AnalysisTask),
+        # which will perform some common setup, including storing:
+        #     self.runDirectory , self.historyDirectory, self.plotsDirectory,
+        #     self.namelist, self.runStreams, self.historyStreams,
+        #     self.calendar, self.namelistMap, self.streamMap, self.variableMap
+        # Here, change MyTask to the name of your task but don't make any other
+        # changes
+        super(MyTask, self).setup_and_check()
+
+        # then, perform additional checks specific to this analysis.
+        # Change or remove these as needed.
+
+        # This checks if myArg was one of the allowed values, and raises an
+        # exception if not
+        possibleArgs = ['blah', 'thing', 'stuff']
+        if self.myArg not in possibleArgs:
+            raise ValueError('MyTask must be constructed with argument myArg '
+                             'having one of the values\n'
+                             '{}.'.format(possibleArgs))
+
```

Comment:
Thanks: the examples below are exactly what I was going to suggest. We can leave setup_and_check as is.